code/modules/mining/equipment/kheiral_cuffs.dm

```diff
@@ -0,0 +1,119 @@
+/**********************Kheiral Cuffs**********************/
+/// Acts as a GPS beacon & connects to station crew monitors from lavaland
+/obj/item/kheiral_cuffs
+	name = "\improper Kheiral cuffs"
+	desc = "A prototype wrist communicator powered by Kheiral Matter. When both ends are clamped to one wrist, acts as a signal range booster for your suit sensors.\n<i>A small engraving on the inside reads, \"NOT HANDCUFFS\"</i>"
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "strand"
+	worn_icon_state = "strandcuff"
+	slot_flags = ITEM_SLOT_GLOVES
+	throwforce = 0
+	w_class = WEIGHT_CLASS_SMALL
+	gender = PLURAL
+	throw_speed = 3
+	throw_range = 5
+	attack_verb_continuous = list("connects")
+	attack_verb_simple = list("connect")
+	/// If we're in the glove slot
+	var/on_wrist = FALSE
+	/// If the GPS is already on
+	var/gps_enabled = FALSE
+	/// If we're off the station's Z-level
+	var/far_from_home = FALSE
+
+/obj/item/kheiral_cuffs/Initialize()
+	. = ..()
+	update_icon(UPDATE_OVERLAYS)
+	RegisterSignal(src, COMSIG_MOVABLE_Z_CHANGED, .proc/check_z)
+
+	check_z(new_turf = loc)
+
+/obj/item/kheiral_cuffs/examine(mob/user)
+	. = ..()
+	if(gps_enabled)
+		. += span_notice("The cuff's GPS signal is on.")
+
+/obj/item/kheiral_cuffs/item_action_slot_check(slot)
+	return slot == ITEM_SLOT_GLOVES
+
+/obj/item/kheiral_cuffs/equipped(mob/user, slot, initial)
+	. = ..()
+	if(slot != ITEM_SLOT_GLOVES)
+		return
+	on_wrist = TRUE
+	playsound(loc, 'sound/weapons/handcuffs.ogg', 30, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	connect_kheiral_network(user)
+
+/obj/item/kheiral_cuffs/dropped(mob/user, silent)
+	. = ..()
+	if(on_wrist)
+		playsound(loc, 'sound/weapons/handcuffs.ogg', 30, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	on_wrist = FALSE
+	remove_kheiral_network(user)
+
+/// Enables the GPS and adds the multiz trait
+/obj/item/kheiral_cuffs/proc/connect_kheiral_network(mob/living/user)
+	if(gps_enabled)
+		return
+	if(!on_wrist || !far_from_home)
+		return
+	var/gps_name = "Unknown"
+	var/obj/item/card/id/id_card = user.get_idcard(hand_first = FALSE)
+	if(id_card)
+		gps_name = id_card.registered_name
+	AddComponent(/datum/component/gps, "*[gps_name]'s Kheiral Link")
+	balloon_alert(user, "GPS activated")
+	ADD_TRAIT(user, TRAIT_MULTIZ_SUIT_SENSORS, src)
+	gps_enabled = TRUE
+
+/// Disables the GPS and removes the multiz trait
+/obj/item/kheiral_cuffs/proc/remove_kheiral_network(mob/user)
+	if(!gps_enabled)
+		return
+	if(on_wrist && far_from_home)
+		return
+	balloon_alert(user, "GPS de-activated")
+	qdel(GetComponent(/datum/component/gps))
+	REMOVE_TRAIT(user, TRAIT_MULTIZ_SUIT_SENSORS, src)
+	gps_enabled = FALSE
+
+/// If we're off the Z-level, set far_from_home = TRUE. If being worn, trigger kheiral_network proc
+/obj/item/kheiral_cuffs/proc/check_z(datum/source, turf/old_turf, turf/new_turf)
+	SIGNAL_HANDLER
+
+	if(!isturf(new_turf))
+		return
+
+	var/mob/living/bridges = loc
```

Comment:
What happens when the cuffs move z levels, but aren't in a living mob?

Reply:
it just changes `far_from_home = TRUE`
I had to put the mob var there because both an if and an else statement use them

Reply:
Won't `connect_kheiral_network` runtime if user is not a human?

Reply:
I wouldn't see a reason for it to fail, it only uses procs from `mob/living`, and to my knowledge the only mobs that're able to put things on their wrists are `mob/living` mobs.

Reply:
It's fixed now, but for the sake of clarity
```
Cuffs are spawned, register to the z level change event
Miner throws cuffs into space, cuffs cross z level
check_z() is called, the cuff's loc is a space tile
space tile is cast to /mob/living, this does nothing
remove_kheiral_network() with the space tile as an arg, lemon dies in a burning fire
```

There's still some weirdness here, see new comment

Reply:
ahhh, right. Thank you for explaining it to me.